Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift

```diff
@@ -0,0 +1,316 @@
+        let result: Result<Networking.InboxNote, Error> = waitFor { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            let action = InboxNotesAction.dismissInboxNote(siteID: self.sampleSiteID, noteID: sampleInboxNoteID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then no inbox notes should be stored
+        XCTAssertEqual(storedInboxNotesCount, 0)
+        XCTAssertFalse(result.isSuccess)
+    }
+
+    func test_dismissInboxNote_then_it_returns_error_upon_empty_response() {
+        // Given an empty network response
+        let sampleInboxNoteID: Int64 = 296
+        XCTAssertEqual(storedInboxNotesCount, 0)
+
+        // When dispatching a `dismissInboxNote` action
+        let result: Result<Networking.InboxNote, Error> = waitFor { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            let action = InboxNotesAction.dismissInboxNote(siteID: self.sampleSiteID, noteID: sampleInboxNoteID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then no inbox notes should be stored
+        XCTAssertEqual(storedInboxNotesCount, 0)
+        XCTAssertFalse(result.isSuccess)
+    }
+
+    func test_markInboxNoteAsActioned_then_it_updates_stored_inbox_notes_and_inbox_action_upon_successful_response() {
+        // Given a stubbed inbox note network response
+        let sampleInboxNoteID: Int64 = 296
+        let sampleActionID: Int64 = 13329
+        let initialInboxNote = sampleInboxNote(id: 296)
+        network.simulateResponse(requestUrlSuffix: "admin/notes/\(sampleInboxNoteID)/action/\(sampleActionID)", filename: "inbox-note")
+        storageManager.insertSampleInboxNote(readOnlyInboxNote: initialInboxNote)
+        XCTAssertEqual(storedInboxNotesCount, 1)
+
+        // When dispatching a `markInboxNoteAsActioned` action
+        let result: Result<Networking.InboxNote, Error> = waitFor { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            let action = InboxNotesAction.markInboxNoteAsActioned(siteID: self.sampleSiteID, noteID: sampleInboxNoteID, actionID: sampleActionID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then the initial inbox action should have it's values updated
+        let updatedInboxNote = viewStorage.loadInboxNote(siteID: sampleSiteID, id: 296)
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertNotEqual(initialInboxNote.actions.first?.name, updatedInboxNote?.actions?.first?.name)
+        XCTAssertNotEqual(initialInboxNote.actions.first?.label, updatedInboxNote?.actions?.first?.label)
+        XCTAssertNotEqual(initialInboxNote.actions.first?.status, updatedInboxNote?.actions?.first?.status)
```

Comment:
nit: is it possible to assert with the expected value after a note is marked as actioned, like the `status` from `unactioned` to `actioned`?